### PR TITLE
Fix saliency map issues

### PR DESCRIPTION
**Pull request by eugene123tw**
Related PRs:
* MPA: https://github.com/openvinotoolkit/model_preparation_algorithm/pull/51
* mmseg: https://github.com/openvinotoolkit/mmsegmentation/pull/93
* mmdet: https://github.com/openvinotoolkit/mmdetection/pull/357

### CODE REVIEW

**Comment by eugene123tw** [comment]
run ote-test

**Review by harimkang — DISMISSED** [review]
Is this an issue that didn't work because the type and name of ResultMediaEntity were changed?

File: external/mmsegmentation/segmentation_tasks/apis/segmentation/openvino_task.py [hunk 1]
```diff
@@ -164,25 +164,36 @@ def infer(self,
...
+                    if label_index == 0:
+                        continue
+                    if len(soft_prediction.shape) == 3:
+                        current_label_soft_prediction = soft_prediction[:, :, label_index]
+                    else:
+                        current_label_soft_prediction = soft_prediction
+                    class_act_map = get_activation_map(current_label_soft_prediction)
+                    result_media = ResultMediaEntity(name=f'{label.name}',
+                                                     type='Soft Prediction',
+                                                     label=label,
```
Comment (goodsong81) [inline external/mmsegmentation/segmentation_tasks/apis/segmentation/openvino_task.py]: 'name' & 'type' convention is inconsistent compared to mmseg pytorch task.

File: external/mmsegmentation/segmentation_tasks/apis/segmentation/model_wrappers/blur.py [hunk 2]
```diff
@@ -78,18 +81,16 @@ def postprocess(self, outputs: Dict[str, np.ndarray], metadata: Dict[str, Any]):
...
-        
-        if 'feature_vector' not in outputs or 'saliency_map' not in outputs:
-            warnings.warn('Could not find Feature Vector and Saliency Map in OpenVINO output. '
-                'Please rerun OpenVINO export or retrain the model.')
-            metadata["saliency_map"] = None
+        hard_prediction = cv2.resize(hard_prediction, metadata['original_shape'][1::-1], 0, 0,
+                                     interpolation=cv2.INTER_NEAREST)
+        soft_prediction = cv2.resize(soft_prediction, metadata['original_shape'][1::-1], 0, 0,
+                                     interpolation=cv2.INTER_NEAREST)
+        metadata['soft_prediction'] = soft_prediction
```
Comment (goodsong81) [inline external/mmsegmentation/segmentation_tasks/apis/segmentation/model_wrappers/blur.py:88]: Maybe iterate for per-label soft prediction maps?

**Review by goodsong81 — CHANGES_REQUESTED** [review]

File: external/mmsegmentation/segmentation_tasks/apis/segmentation/model_wrappers/blur.py (hunk 2 above)
Comment (eugene123tw) [reply external/mmsegmentation/segmentation_tasks/apis/segmentation/model_wrappers/blur.py:88]: `soft_prediction` includes soft prediction for each label as the tensor dimension is (H, W, Number of classes + 1). The iteration occurs here in `infer`: 
```
for label_index, label in self._label_dictionary.items():
      if label_index == 0:
          continue
      if len(soft_prediction.shape) == 3:
          current_label_soft_prediction = soft_prediction[:, :, label_index]
      else:
          current_label_soft_prediction = soft_prediction
      class_act_map = get_activation_map(current_label_soft_prediction)
      result_media = ResultMediaEntity(name=f'{label.name}',
                                       type='Soft Prediction',
                                       label=label,
                                       annotation_scene=dataset_item.annotation_scene,
                                       roi=dataset_item.roi,
                                       numpy=class_act_map)
      dataset_item.append_metadata_item(result_media, model=self.model)
```

File: external/mmsegmentation/segmentation_tasks/apis/segmentation/openvino_task.py (hunk 1 above)
Comment (eugene123tw) [reply external/mmsegmentation/segmentation_tasks/apis/segmentation/openvino_task.py]: fixed

**Review by harimkang — APPROVED** [review]

**Comment by harimkang** [comment]
I'm sorry. I'll look at the submodule first and approve it again.

**Review by harimkang — CHANGES_REQUESTED** [review]
I'm sorry. I'll look at the submodules first, and I'll approve it again after the commit of the submodules are merged and changed.

**Review by harimkang — DISMISSED** [review]
It seems to me that there is still a conflict on the submodule side.

**Review by harimkang — DISMISSED** [review]

File: external/mmsegmentation/segmentation_tasks/apis/segmentation/inference_task.py [hunk 3]
```diff
@@ -223,18 +222,26 @@ def _add_predictions_to_dataset(self, prediction_results, dataset):
...
-                         model: torch.nn.Module, config: Config, dataset: DatasetEntity,
-                         dump_features: bool = False, dump_saliency_map: bool = False) -> None:
+            if dump_soft_prediction:
+                for label_index, label in self._label_dictionary.items():
+                    if label_index == 0:
+                        continue
+                    if len(soft_prediction.shape) == 3:
+                        current_label_soft_prediction = soft_prediction[:, :, label_index]
+                    else:
+                        current_label_soft_prediction = soft_prediction
```
Comment (goodsong81) [inline external/mmsegmentation/segmentation_tasks/apis/segmentation/inference_task.py]: Could you explain this branching logic? Do we have dim != 3 cases?
We see `soft_prediction = np.transpose(prediction, axes=(1, 2, 0)` assuming 3 dims.

Comment (eugene123tw) [reply external/mmsegmentation/segmentation_tasks/apis/segmentation/inference_task.py]: Fixed

**Review by goodsong81 — APPROVED** [review]
LGTM!

**Review by chuneuny-emily — APPROVED** [review]
